lib/puppet/transaction/report.rb

```diff
@@ -70,6 +70,10 @@ class Puppet::Transaction::Report
   #
   attr_reader :metrics
 
+  # A hash of number of changes per resource
+  # @return [Hash<{String => Hash<{String => Event Status}>}] A map of type+title to number of events
+  attr_accessor :changes
```

Comment:
Do we actually need an attribute here?  Seems like there could be a single method called `display_changes` that "sumups" and then displays them.